src/main/java/org/elasticsearch/search/aggregations/AggregatorFactories.java

```diff
@@ -49,8 +44,8 @@ private AggregatorFactories(AggregatorFactory[] factories) {
         this.factories = factories;
     }
 
-    private static Aggregator createAndRegisterContextAware(AggregationContext context, AggregatorFactory factory, Aggregator parent, long estimatedBucketsCount) {
-        final Aggregator aggregator = factory.create(context, parent, estimatedBucketsCount);
+    private static Aggregator createAndRegisterContextAware(AggregationContext context, AggregatorFactory factory, Aggregator parent, boolean collectsOnly0) throws IOException {
+        final Aggregator aggregator = factory.create(context, parent, collectsOnly0);
```

Comment:
I wonder if something like `collectsSingleBucket` would be better instead of `collectsOnly0`?